test/runtests.jl

```diff
@@ -216,6 +216,46 @@ end
 
 rm(joinpath(Documenter_root, "build"); recursive = true)
 
+# Try generating documentation with generate()
+
+pkg_generate = if VERSION >= v"0.5-"
+    Pkg.add("PkgDev")
```

Comment:
It depends how many packages you've got, how recently `Pkg.update` was run, and the state of repos. Can take a while sometimes.

If `PkgDev` was installable in `0.4` then we could add it to a `test/REQUIRE` file, but as far as I know it's limited to `0.5` since it requires `Base.LibGit2`. The way you've done it would be how I would have do it as well.